cpp/src/arrow/flight/types.cc

```diff
@@ -21,6 +21,7 @@
 #include <sstream>
 #include <utility>
 
+#include "arrow/buffer.h"
```

Comment:
Is this header used?

Reply:
It's actually needed for the descriptor->size() call below.